### PR TITLE
dqm plot, pixel endcap roc occupancy in 81x

### PR DESCRIPTION
**Pull request by gouskos**

### CODE REVIEW

**Comment by boudoul** [comment]
hello @dmitrijus , could you please trigger  the tests ? Thanks

**Comment by dmitrijus** [comment]
+1

**Comment by dmitrijus** [comment]
please test

File: DQM/SiPixelMonitorTrack/src/SiPixelTrackResidualSource.cc [hunk 1]
```diff
@@ -1124,6 +1136,40 @@ void SiPixelTrackResidualSource::analyze(const edm::Event& iEvent, const edm::Ev
...
+                int clu_roc_biny = -99.;
+                int nrocly = pxfmodule + pxfpanel;
+                for (int i=0; i<nrocly; i++) {
+                  int j = (pxfdisk<0) ? i : nrocly - 1 - i;
+                  if (yclust>=(j*52.0)&& yclust<((j+1)*52.0))
+                    clu_roc_biny = 6 - nrocly + 2 * i + ((pxfblade>0) ? pxfblade-1 : pxfblade + 12)*12 + 1;
+                }
+                if (pxfblade>0) { clu_roc_biny = clu_roc_biny+144; }
+
+                meRocBladevsDiskEndcap->setBinContent(clu_roc_binx,clu_roc_biny, meRocBladevsDiskEndcap->getBinContent(clu_roc_binx,clu_roc_biny)+1);
```
Comment (davidlange6) [inline DQM/SiPixelMonitorTrack/src/SiPixelTrackResidualSource.cc]: @gouskos - is this not the same as meRocBladevsDiskEndcap->Fill(....)?

Comment (davidlange6) [inline DQM/SiPixelMonitorTrack/src/SiPixelTrackResidualSource.cc]: @gouskos - to answer my question, "no" - but the appropriate function call is meRocBladevsDiskEndcap->AddBinContent(clu_roc_binx,clu_roc_biny)
could you have a look. Thanks

**Comment by gouskos** [comment]
Ok - I will update the code asap

File: DQM/SiPixelMonitorTrack/src/SiPixelTrackResidualSource.cc (hunk 1 above)
Comment (gouskos) [inline DQM/SiPixelMonitorTrack/src/SiPixelTrackResidualSource.cc]: Hi, there is no function "AddBinContent" for the MonitorElement class.

Comment (fioriNTU) [inline DQM/SiPixelMonitorTrack/src/SiPixelTrackResidualSource.cc]: @gouskos if you use the correct function from here http://cmslxr.fnal.gov/source/DQMServices/Core/interface/MonitorElement.h#0337 you can get the root object (TH2) and use the AddBinContent function on it. Hope this helps.

**Comment by gouskos** [comment]
@davidlange6  , @dmitrijus I tried following your suggestion by implementing the code below:

Int_t bin  = meRocBladevsDiskEndcap->getTH2F()->GetBin(clu_roc_binx,clu_roc_biny);
Int_t binp = meRocBladevsDiskEndcap->getTH2F()->GetBin(clu_roc_binx,clu_roc_biny+1);
meRocBladevsDiskEndcap->getTH2F()->AddBinContent(bin,1);
meRocBladevsDiskEndcap->getTH2F()->AddBinContent(binp,1);    

However the histograms are not filled. I spent some time debugging this but without success. Given that this PR is quite urgent for the tracker, you it be possible to consider staying with the initial version of the code and then try to improve the code with your suggestions? The same holds also for the 80X version of this PR [15236]

**Comment by fioriNTU** [comment]
@davidlange6 and @dmitrijus I confirm what @gouskos said, we had a try to implement the changes suggested by David but it doesn't seem to work properly. If you think that what we did is wrong please give a suggestion on what should be changed. However this PR is quite urgent to have merged, it will allow to build a monitor of inefficienct ROCS in Pixels. I would like this to be merged and we can continue searching for a more efficient way to do the same thing.

**Comment by davidlange6** [comment]
+1

weird - maybe a bug in root - will try to test

**Pull request merged**